src/testRunner/unittests/tsbuild/javascriptProjectEmit.ts

```diff
@@ -0,0 +1,284 @@
+                        },
+                        "include": ["nominal.js"]
+                    }`,
+                "/src/sub-project/index.js": utils.dedent`
+                    /**
+                     * @typedef {Nominal<string, 'MyNominal'>} MyNominal
+                     */
+                    const c = /** @type {*} */(null);
+                    `,
+                "/src/sub-project/tsconfig.json": utils.dedent`
+                    {
+                        "extends": "../tsconfig.base.json",
+                        "compilerOptions": {
+                            "composite": true,
+                            "outFile": "sub-project.js"
+                        },
+                        "references": [
+                            { "path": "../common", "prepend": true }
+                        ],
+                        "include": ["./index.js"]
+                    }`,
+                "/src/sub-project-2/index.js": utils.dedent`
+                    const variable = {
+                        key: /** @type {MyNominal} */('value'),
+                    };
+
+                    /**
+                     * @return {keyof typeof variable}
+                     */
+                    function getVar() {
+                        return 'key';
+                    }
+                    `,
+                "/src/sub-project-2/tsconfig.json": utils.dedent`
+                    {
+                        "extends": "../tsconfig.base.json",
+                        "compilerOptions": {
+                            "composite": true,
+                            "outFile": "sub-project-2.js"
+                        },
+                        "references": [
+                            { "path": "../sub-project", "prepend": true }
+                        ],
+                        "include": ["./index.js"]
+                    }`,
+                "/src/tsconfig.json": utils.dedent`
+                    {
+                        "compilerOptions": {
+                            "composite": true,
+                            "outFile": "src.js"
+                        },
+                        "references": [
+                            { "path": "./sub-project", "prepend": true },
+                            { "path": "./sub-project-2", "prepend": true }
+                        ],
+                        "include": []
+                    }`,
+                "/src/tsconfig.base.json": utils.dedent`
+                    {
+                        "compilerOptions": {
+                            "skipLibCheck": true,
+                            "rootDir": "./",
+                            "allowJs": true,
+                            "checkJs": true,
+                            "declaration": true
+                        }
+                    }`,
+            }, symbolLibContent);
+        });
+        after(() => {
+            projFs = undefined!;
+        });
+        verifyTsc({

```

Comment:
I think you want to use `verifyTscIncrementalEdits` passing in   `incrementalScenarios` as array with literal containing buildKind and modifyFs you have in next tsc command, so that it first builds this scenario and then makes changes and builds incremental edit scenario. (currently your next test case is running as if the initial fs before any build had the changes per modifyFs).